### DISCUSSION TITLE
Access Query Strings from the URL when using the Webhook node?

### DISCUSSION THREAD

**Discussion by Adman1020** [zachowj/hass-node-red#50] [Q&A]
Is it possible to access and use the query strings passed to a webhook within the URL? For example, being able to get the abc123 from POST http://<my-ha-url>.duckdns.org/api/webhook/test?testQuery=123abc.

Only the Header and Payload seem to be exposed by the node?

**Comment by zachowj** [comment]
Currently, no. 

I think typically using a POST method you would use the payload to submit data, but I see how it could be useful if you didn't have control over what is being posted.

**Reply by Adman1020** [reply]
Indeed. Annoying the API I'm trying to setup a subscription to requires you to return a token that is sent as a query string in order to validate the endpoint.

**Reply by zachowj** [reply]
If you're not using the header output for any of your flows. A quick and dirty solution would be to replace the header output with the URL and you could parse the query string out.

https://github.com/zachowj/hass-node-red/blob/88a58baa7b32fdea17eaa7afab94d134a68fd7b5/custom_components/nodered/websocket.py#L125

```python
data = {"payload": payload, "headers": str(request.url)}
```

**Reply by Adman1020** [reply]
Cheers. I don't know if I will need headers yet, its early days. So I have added the URL to the existing headers with the below for the time being:

```
headers = dict(request.headers)
headers['Url'] = str(request.url)
data = {"payload": payload, "headers": headers}
```
